x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_edit.test.tsx

```diff
@@ -0,0 +1,159 @@
+      iconClass: 'test',
+      selectMessage: 'test',
+      validateConnector: (): ValidationResult => {
+        return { errors: {} };
+      },
+      validateParams: (): ValidationResult => {
+        const validationResult = { errors: {} };
+        return validationResult;
+      },
+      actionConnectorFields: null,
+      actionParamsFields: null,
+    };
+
+    const alert = {
+      id: 'ab5661e0-197e-45ee-b477-302d89193b5e',
+      params: {
+        aggType: 'average',
+        threshold: [1000, 5000],
+        index: 'kibana_sample_data_flights',
+        timeField: 'timestamp',
+        aggField: 'DistanceMiles',
+        window: '1s',
+        comparator: 'between',
+      },
+      consumer: 'alerting',
+      alertTypeId: 'my-alert-type',
+      enabled: false,
+      schedule: { interval: '1m' },
+      actions: [
+        {
+          actionTypeId: 'my-action-type',
+          group: 'threshold met',
+          params: { message: 'Alert [{{ctx.metadata.name}}] has exceeded the threshold' },
+          message: 'Alert [{{ctx.metadata.name}}] has exceeded the threshold',
+          id: '917f5d41-fbc4-4056-a8ad-ac592f7dcee2',
+        },
+      ],
+      tags: [],
+      name: 'Serhii',
+      throttle: null,
+      apiKeyOwner: null,
+      createdBy: 'elastic',
+      updatedBy: 'elastic',
+      createdAt: new Date(),
+      muteAll: false,
+      mutedInstanceIds: [],
+      updatedAt: new Date(),
+    };
+    actionTypeRegistry.get.mockReturnValueOnce(actionTypeModel);
+    actionTypeRegistry.has.mockReturnValue(true);
+    alertTypeRegistry.list.mockReturnValue([alertType]);
+    alertTypeRegistry.get.mockReturnValue(alertType);
+    alertTypeRegistry.has.mockReturnValue(true);
+    actionTypeRegistry.list.mockReturnValue([actionTypeModel]);
+    actionTypeRegistry.has.mockReturnValue(true);
+
+    await act(async () => {
+      wrapper = mountWithIntl(
+        <AlertsContextProvider
+          value={{
+            reloadAlerts: () => {
+              return new Promise<void>(() => {});
+            },
+            http: deps!.http,
+            actionTypeRegistry: deps!.actionTypeRegistry,
+            alertTypeRegistry: deps!.alertTypeRegistry,
+            toastNotifications: deps!.toastNotifications,
+            uiSettings: deps!.uiSettings,
+          }}
+        >
+          <AlertEdit
+            editFlyoutVisible={true}
+            setEditFlyoutVisibility={state => {}}
+            initialAlert={alert}
+          />
+        </AlertsContextProvider>
+      );
+    });
+    await waitForRender(wrapper);
+  });
+
+  it('renders alert add flyout', () => {
+    expect(wrapper.find('[data-test-subj="editAlertFlyoutTitle"]').exists()).toBeTruthy();
+    expect(wrapper.find('[data-test-subj="saveEditedAlertButton"]').exists()).toBeTruthy();
+  });
+});
+
+async function waitForRender(wrapper: ReactWrapper<any, any>) {
+  await Promise.resolve();
+  await Promise.resolve();
+  wrapper.update();
```

Comment:
This feels wrong... if a test is relying on this to pass it must be weirdly fragile - I'd be happy to jump on a call so we can talk this through, maybe we'll find a more maintainable approach.